js/filters.js

```diff
@@ -143,14 +143,22 @@ function TypeaheadFilter(elm) {
   Filter.call(this, elm);
 
   var key = this.$body.data('dataset');
+  var allowText = this.$body.data('allow-text') !== undefined;
   var dataset = typeahead.datasets[key];
-  this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset);
+  this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset, allowText);
```

Comment:
The intention here is that `data-allow-text` is either present or absent:

```
<div data-allow-text>text allowed</div>
<div>text not allowed</div>
```

The `allowText` variable in the javascript is meant to be a boolean flag that signals whether `data-allow-text` is set or not.